lib/classes/media/subdef.php

```diff
@@ -362,11 +362,12 @@ public function get_record_id()
 
     public function getEtag()
     {
-        if (!$this->etag && $this->is_physically_present()) {
-            $file = new SplFileInfo($this->getRealPath());
-
+        $file = new SplFileInfo($this->getRealPath());
+        // etag is generated from md5 of path and file last modified time
+        // so check if etag is to be changed
+        if ((!$this->etag && $this->is_physically_present()) || $this->etag != md5($file->getRealPath() . $file->getMTime())) {
```

Comment:
`md5($file->getRealPath() . $file->getMTime()) `is the same value as `generateEtag()`
so in fact we do `if(x != f()) then x=f()` --> `x=f()` if faster